### PR TITLE
DM-27467: Document DDS "Lazy master" problem in tstn-023

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -536,7 +597,12 @@ After the initial test it seemed like the Jupyter notebooks from the nublado pla
 During the tests of the features introduced above it was possible to demonstrate that the single process mode is now capable of joining a domain and communicating with the CSCs in the system.
 In particular, the feature that seems to allow the use of single process mode is the introduction of the :ref:`DDS data partitioning <Implementing-a-new-DDS-data-partitioning-schema>`.
 
-This considerably simplify handling the nublado configuration as the containers can continue to be hosted in public repositories, there is no need to configure them to use shared memory and to worry about users notebooks crashing and corrupting the daemons they are attached to, potentially affecting other systems.
+Although single process from notebooks **can** work with a large scale system deployed with shared memory, we did noticed some small issues.
```
Comment (mareuter) [inline index.rst]: Change "noticed" to "notice".

File: index.rst [hunk 2]
```diff
@@ -527,6 +527,67 @@ Ultimately, that means all daemons and applications running in single process mo
...
+DDS Master
+----------
+
+When deploying several applications in different nodes (also called federations), DDS will select a node to serve as the master of the system.
+The master is in charge of keeping a dictionary of all the topics in the system and its durability service will work as the source of historical data for new applications.
+In a sense, the master node acts like a broker for historical data.
+
+The process of selecting a master for the system is handled by DDS using the ``masterPriority`` for each application, which is configured via ``NameSpaces/Policy/masterPriority`` entry in the ospl configuration file.
+
+The selection process is explained in details in the `master priority`_ section of the `ospl deployment guide`_.
```
Comment (mareuter) [inline index.rst]: Say "explained in the" instead of "explained in details in the".

File: index.rst [hunk 3]
```diff
@@ -527,6 +527,67 @@ Ultimately, that means all daemons and applications running in single process mo
...
+Basically, the system with the highest value of ``masterPriority`` is selected as the master.
+If all nodes have the same ``masterPriority``, then the one with the highest namespace quality (measured by DDS) is selected and, if a tie still exists, the system with highest ID is selected.
+
+.. _master priority: http://download.prismtech.com/docs/Vortex/html/ospl/DeploymentGuide/guide.html#masterpriority
+.. _ospl deployment guide: http://download.prismtech.com/docs/Vortex/html/ospl/DeploymentGuide/guide.html
+
+Some important notes to keep in mind:
+  - The value of ``masterPriority`` must be between 0 and 254.
```
Comment (mareuter) [inline index.rst]: Technically the high end if the range is 255.

File: index.rst [hunk 4]
```diff
@@ -527,6 +527,67 @@ Ultimately, that means all daemons and applications running in single process mo
...
+One thing we noticed in the system is that the HRTO_ would usually be connected to issues with the master node.
+In some cases, the selected master would become too busy to be able to handle the alignment process which would cause the historical read alignment to either take longer or fail altogether.
+In order to minimize this problem we now carefully select a daemon to be the master, and make sure nothing else is attached to it.
+
+In general, the master daemon is configured with ``OSPL_MASTER_PRIORITY=201`` and the other daemons are set with ``OSPL_MASTER_PRIORITY=5`` or ``OSPL_MASTER_PRIORITY=10``.
+To ensure the single process nodes will not attempt to become masters we recommend setting ``OSPL_MASTER_PRIORITY=0``.
+
+.. _Namespace policy:
+
+Namespace alinee
```
Comment (mareuter) [inline index.rst]: This should be alignee not alinee. This is repeated many times below.

File: index.rst [hunk 5]
```diff
@@ -527,6 +527,67 @@ Ultimately, that means all daemons and applications running in single process mo
...
+----------------
+
+The durability service on all nodes are designed to store historical data for all applications in the system.
+Nevertheless, one can anticipate that this behavior would not scale for large scale systems.
```
Comment (mareuter) [inline index.rst]: Second scale feels redundant, recommend deleting it.

File: index.rst [hunk 6]
```diff
@@ -527,6 +527,67 @@ Ultimately, that means all daemons and applications running in single process mo
...
+In fact, we have observed problems with this procedure in our systems as single process nodes are simply not capable of keeping up with the entire system traffic.
+
+DDS allow one to configure this behavior on the durability services by means of the `alinee policy`_.
```
Comment (mareuter) [inline index.rst]: "allows" instead of "allow".

File: index.rst [hunk 7]
```diff
@@ -536,7 +597,12 @@ After the initial test it seemed like the Jupyter notebooks from the nublado pla
 During the tests of the features introduced above it was possible to demonstrate that the single process mode is now capable of joining a domain and communicating with the CSCs in the system.
 In particular, the feature that seems to allow the use of single process mode is the introduction of the :ref:`DDS data partitioning <Implementing-a-new-DDS-data-partitioning-schema>`.
 
-This considerably simplify handling the nublado configuration as the containers can continue to be hosted in public repositories, there is no need to configure them to use shared memory and to worry about users notebooks crashing and corrupting the daemons they are attached to, potentially affecting other systems.
+Although single process from notebooks **can** work with a large scale system deployed with shared memory, we did noticed some small issues.
+Probably the most critical one is that creating a salobj Remote soon after creating a Domain will, in most cases, cause a HRTO_ problem.
+After inspecting the DDS logs it was possible to track this issue to trying to start DDS readers while the domain was still aligning with the other systems.
```
Comment (mareuter) [inline index.rst:602]: Is this where the 10 second delay after creating a domain recommendation comes from while we have nublado in single process mode?

**Review by mareuter — APPROVED** [review]
Mostly looks good. Some wordsmithing and spelling to fix. The one question can be addressed without holding up fixing and merging the text.

File: index.rst (hunk 7 above)
Comment (tribeiro) [reply index.rst:602]: Exactly...

**Pull request merged**